draft-dnsop-deleg-transport.md

```diff
@@ -100,76 +100,82 @@ contributor:
+resolvers.  Note that these could potentially be the same addresses as
+are used in the hints of a DELEG record; NS and DELEG are orthogonal.
 
-The example.net authoritative server may return the following SVCB records in response to a query as directed by the above records.
+    ns1.example.com. 86400   IN  A      192.0.2.1
+    ns1.example.com  86400   IN  AAAA   2001:DB8::1
 
-    config2.example.net 3600    IN SVCB ns2.example.net. (
-                    alpn=dot
-                    tlsa="2 0 1 ABC..."
-                    ipv4hint=192.0.2.54
-                    ipv6hint=2001:db8:2423::3 )
+As with SVCB, DELEG also offer the ability to use AliasMode for a
+delegation. The example below shows that example.com is
+being delegated with a DELEG AliasMode record which can then be
+further resolved using SVCB to locate the actual parameters.
 
-The above records indicate to the client that the actual configuration for the example.com zone can be found at config2.example.net.
+    example.com.  86400  IN DELEG 0 config2.example.net.
+    example.com.  86400  IN NS      ns1.example.net.
 
-Later sections of this document will go into more detail on the resolution process using these records.
+The example.net authoritative server's SVCB record could look like this:
 
-## Goal of the DELEG record
+    config2.example.net 3600    IN SVCB ns2.example.net. (
+                        alpn=dot
+                        tlsa="2 0 1 ABC..."
+                        ipv4hint=192.0.2.54
+                        ipv6hint=2001:db8:2423::3 )
 
-The primary goal of transport specification in DELEG records is to provide zone owners a way to signal to new clients how to connect to servers serving a child domain that can coexist with NS records in the same zone, and do not break software that does not support new transports.
+This document defines in detail the resolution process using these records.
 
-### SvcParams
+## SvcParams
 
-All SvcParamKeys for the "dns" scheme {{?9461}} apply as specified.  These are the "transport parameters", describing how to reach an endpoint.
+All SvcParamKeys for the "dns" scheme from SVCB-DNS {{?RFC9461}} apply as specified.  These are the "transport parameters", attributes describing how to reach an endpoint.
 
-The "alpn" transport parameter is OPTIONAL to include (unlike in SVCB-DNS, where it is generally required).  If the "alpn" SvcParamKey is omitted, the only available transport is presumed to be unencrypted DNS over UDP/TCP port 53.  Endpoints can indicate that insecure transport is not available by specifying "mandatory=alpn".
+The "alpn" transport parameter is OPTIONAL to include, unlike in SVCB-DNS, where it is generally required.  If the "alpn" SvcParamKey is omitted, the only available transport is presumed to be unencrypted DNS over UDP/TCP port 53.  Endpoints can indicate that insecure transport is not available by specifying "mandatory=alpn". [Why do we not require do53 to be explicit?]
```

Comment:
Are you arguing that Do53 should be opt-in instead of opt-out?  We can certainly structure it that way; the difference seems pretty superficial.  I appreciate this approach as an application of Occam's Razor, but it's possible that "mandatory=alpn" is too obscure or is stretching the "mandatory" semantics too far.

Reply:
Yes, I am arguing that if you are using a DELEG record, it should be explicit when you want do53 with it.  I don't want really anything assumed.  

It doesn't have to be in this draft initially though, since I can argue my case on the list.

(If all you wanted was Legacy DNS then just NS records.)

Reply:
```suggestion
The "alpn" transport parameter is OPTIONAL to include, unlike in SVCB-DNS, where it is generally required.  If the "alpn" SvcParamKey is omitted, the only available transport is presumed to be unencrypted DNS over UDP/TCP port 53.  Endpoints can indicate that insecure transport is not available by specifying "mandatory=alpn". [Should we change this to make do53 require a more explicit opt-in?]
```